logger.go

```diff
@@ -24,37 +32,51 @@ func LogDockerServerInfo(ctx context.Context, client client.APIClient, logger Lo
 }
 
 // TestLogger returns a Logging implementation for testing.TB
-// This way logs from testcontainers are part of the test output of a test suite or test case
+// This way logs from testcontainers are part of the test output of a test suite or test case.
 func TestLogger(tb testing.TB) Logging {
 	tb.Helper()
 	return testLogger{TB: tb}
 }
 
-// WithLogger is a generic option that implements GenericProviderOption, DockerProviderOption
-// It replaces the global Logging implementation with a user defined one e.g. to aggregate logs from testcontainers
-// with the logs of specific test case
+// WithLogger returns a generic option that sets the logger to be used.
+//
+// Consider calling this before other "With functions" as these may generate logs.
+//
+// This can be given a TestLogger to collect the logs from testcontainers into a
+// test case.
 func WithLogger(logger Logging) LoggerOption {
 	return LoggerOption{
 		logger: logger,
 	}
 }
 
+// LoggerOption is a generic option that sets the logger to be used.
+//
+// It can be used to set the logger for providers and containers.
 type LoggerOption struct {
 	logger Logging
 }
 
+// ApplyGenericTo implements GenericProviderOption.
 func (o LoggerOption) ApplyGenericTo(opts *GenericProviderOptions) {
 	opts.Logger = o.logger
 }
 
+// ApplyDockerTo implements DockerProviderOption.
 func (o LoggerOption) ApplyDockerTo(opts *DockerProviderOptions) {
 	opts.Logger = o.logger
 }
 
+// Customize implements ContainerCustomizer.
```

Comment:
Because this option is now available to the modules, we need to document it. You already know how to do it. See https://github.com/testcontainers/testcontainers-go/pull/2260/files#r1498933117

Thanks! 🙏